src/app/screens/confirmStxTransaction/index.tsx

```diff
@@ -123,7 +134,7 @@ function ConfirmStxTransaction() {
     }
 
     const txAmount = new BigNumber(txPayload.amount.toString(10));
-    const txFee = new BigNumber(unsignedTx.auth.spendingCondition.fee.toString());
+    const txFee = new BigNumber(fee.toString() ?? unsignedTx.auth.spendingCondition.fee.toString());
```

Comment:
fee.toString() will always be non null/endefined, so it will never get the fee from the spending condition.